mysqld-exporter/main.libsonnet

```
@@ -43,10 +51,25 @@ local container = k.core.v1.container;
             DATA_SOURCE_NAME: '$(MYSQL_USER):$(MYSQL_PASSWORD)@tcp($(MYSQL_HOST):$(MYSQL_PORT))/?tls=$(MYSQL_TLS_MODE)',
           }),
         ]
-      ),
+      ) + if needsConfigFile then configVolumeMount(configMapName, '/conf') else {},
```

Note that the suggestion might not be `jsonnetfmt` compliant. Wrapping the if/else block in parenthesis removes potential the _footgun_ about extending the code and only applying to one branch of the conditional block:

```suggestion
      ) + (if needsConfigFile then configVolumeMount(configMapName, '/conf') else {}),
```